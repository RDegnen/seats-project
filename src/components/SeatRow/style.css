.seatRow {
  display: flex;
  justify-content: center;
}