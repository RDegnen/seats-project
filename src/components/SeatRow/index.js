import React from 'react'
import PropTypes from 'prop-types'
import { Row } from 'reactstrap'

import { seatPropType } from '../props'
import RowItem from '../RowItem'
import './style.css'

const SeatRow = ({ row, ...rest }) => (
  <Row className="seatRow">
    {row.map(seat => (
      <RowItem
        key={`${seat.seat}${seat.class}`}
        item={seat}
        {...rest}
      />
    ))}
  </Row>
)

SeatRow.propTypes = {
  row: PropTypes.arrayOf(seatPropType).isRequired,
}

export default SeatRow