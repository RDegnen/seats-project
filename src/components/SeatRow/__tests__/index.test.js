import React from 'react'
import { shallow } from 'enzyme'

import SeatRow from '../'

const row = [ 
  { seat: 'A', row: 3, class: 'First', occupied: true },
  { seat: 'B', row: 3, class: 'First', occupied: true },
]

const getComponent = props => shallow(
  <SeatRow row={row} selectSeat={jest.fn()} {...props} />
)

describe('<SeatRow />', () => {
  it('should match snapshot', () => {
    const component = getComponent()
    expect(component).toMatchSnapshot()
  })
})