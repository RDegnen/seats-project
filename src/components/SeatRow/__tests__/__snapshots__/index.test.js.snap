// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SeatRow /> should match snapshot 1`] = `
<Row
  className="seatRow"
  tag="div"
>
  <RowItem
    item={
      Object {
        "class": "First",
        "occupied": true,
        "row": 3,
        "seat": "A",
      }
    }
    key="AFirst"
    selectSeat={[MockFunction]}
  />
  <RowItem
    item={
      Object {
        "class": "First",
        "occupied": true,
        "row": 3,
        "seat": "B",
      }
    }
    key="BFirst"
    selectSeat={[MockFunction]}
  />
</Row>
`;
